Composite/Plugins/Functions/FunctionProviders/StandardFunctionProvider/Pages/GetPageIdFunction.cs

```diff
@@ -70,30 +71,45 @@ private Guid GetCurrentPageId()
         private Guid? GetCurrentPageIdFromHttpContext()
         {
             var entityToken = HttpContext.Current?.Items[ActionExecutorFacade.HttpContextItem_EntityToken];
-            return GetPageIdFromDataToken(entityToken as DataEntityToken);
+            return GetPageIdFromDataToken(entityToken);
         }
 
         private Guid? GetCurrentPageIdFromFormFlow()
         {
             var currentFormTreeCompiler = FormFlowUiDefinitionRenderer.CurrentFormTreeCompiler;
             if (currentFormTreeCompiler.BindingObjects.TryGetValue(FormsWorkflow.EntityTokenKey, out var entityToken))
             {
-                return GetPageIdFromDataToken(entityToken as DataEntityToken);
+                return GetPageIdFromDataToken(entityToken);
             }
 
             return null;
         }
 
-        private Guid? GetPageIdFromDataToken(DataEntityToken entityToken)
+        private Guid? GetPageIdFromDataToken(object entityTokenObj)
         {
-            if (entityToken == null)
+            if (entityTokenObj is null || !(entityTokenObj is EntityToken entityToken))
+            {
                 return null;
+            }
 
-            if (Type.GetType(entityToken.Type, throwOnError: false) != typeof(IPage))
-                return null;
+            Guid? pageId = null;
 
-            var data = entityToken.Data as IPage;
-            var pageId = data?.Id;
+            if (entityTokenObj is DataEntityToken dataEntityToken)
+            {
+                if (dataEntityToken.Data is IPage cPage)
+                {
+                    pageId = cPage?.Id;
+                }
+                else if (dataEntityToken.Data is IPageRelatedData cPagerelatedDate)
+                {
+                    pageId = cPagerelatedDate?.PageId;
+                }
+            }
+            else if (typeof(IPage).IsAssignableFrom(Type.GetType(entityToken.Type, throwOnError: false)))

```

Comment:
Can you describe the sample for this?

Reply:
Yes, sure

- `if (dataEntityToken.Data is IPage page)` - appears while adding a metadata element;
- `else if (dataEntityToken.Data is IPageRelatedData pageRelatedData)` - appears while **editing** a datafolder element;
- `else if (typeof(IPage).IsAssignableFrom(Type.GetType(entityToken.Type, throwOnError: false)))` - appears while **adding** a datafolder element.

I added explaining comments to original code